test/functionalTest/cases/970_create_entity/creation_with_attribute_as_object.test

```diff
@@ -0,0 +1,111 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+POST Entities V2 with attribute as object
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0
+
+--SHELL--
+
+#
+# 01. POST /v2/entities
+# 02. GET /v2/entities
+#
+
+echo "01. POST /v2/entities"
+echo "===================="
+payload='{
+  "id": "E1",
+  "type": "T1",
+  "attr01": "1",
+  "attr02": 2,
+  "attr03": 3.3,
+  "attr04": true,
+  "attr05": false,
+  "attr06": {
+    "value": "06"
+  },
+  "attr07": {
+    "value": 7
+  },
+  "attr08": {
+    "value": true
+  },
+  "attr09": {
+    "value": 9,
+    "type": "number"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload" --json
+echo
+echo
+
+
+echo "02. GET /v2/entities/E1"
+echo "======================="
+orionCurl --url /v2/entities/E1 --json
+echo
+echo
+
+
+--REGEXPECT--
+01. POST /v2/entities
+====================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E1
+Date: REGEX(.*)
+
+
+
+02. GET /v2/entities/E1
+=======================
+HTTP/1.1 200 OK
+Content-Length: 191
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "attr01": "1", 
+    "attr02": 2.0, 
+    "attr03": 3.2999999999999998, 
```

Comment:
Yes, I know how to do it ;-). After a million times I do it in my sleep ... :-)

I just tested and I see 3.3 in mongo. Not 2.2999...

That is good, at least. I imagine the rounding error is in the last step (snprintf to a string in the render phase). We need to investigate and see if we can do anything about it.

Issue created: #1073